src/runner.cpp

```diff
@@ -31,11 +32,48 @@
 namespace libsemigroups {
   namespace py = pybind11;
 
+  namespace {
+    using std::chrono::high_resolution_clock;
+    using std::chrono::system_clock;
+    using std::chrono::time_point_cast;
+    system_clock::time_point to_system(high_resolution_clock::time_point tp) {
+      if constexpr (std::is_same_v<high_resolution_clock, system_clock>) {
+        return time_point_cast<system_clock::duration>(tp);
+      } else {
+        // Account for the difference between high_resolution_clock and
+        // system_clock
+        auto sys_now   = system_clock::now();
+        auto clock_now = high_resolution_clock::now();
+        return time_point_cast<system_clock::duration>(tp - clock_now
+                                                       + sys_now);
+      }
+    }
```

Comment:
This is what I would like to do, but this causes a compiler error on Mac because:
```
no viable conversion from returned value of type 'time_point<std::chrono::steady_clock, [...]>' to function return type 'time_point<system_clock, [...]>'
```
on line 41, even though the `if constexpr` should prevent this line from being compiled (or so I thought).

Reply:
I understand why this isn't compiling now, but still struggling to get the behaviour I desire. I will persevere

Reply:
The reason it wasn't compiling was because my constexpr if was not inside a template. Therefore, each branch was fully checked by the compiler, which resulted in errors. I think I've fixed this now in 08d4a65.